api-doc/openapi.yaml

```
@@ -1,48 +1,619 @@
 openapi: 3.0.3
 info:
-  version: 0.0.1
+  version: 0.0.4
   title: Vauhtijuoksu API
 paths:
-  /target:
+  /donations:
     get:
+      description: >
+        Get all donations with one request. List of individual donations with
+        metadata.
       responses:
-        200:
-          $ref: '#/components/responses/TargetOk'
+        '200':
+          $ref: '#/components/responses/DonationsAllOk'
+    post:
+      description: Post donation, id generated by backend..
+      security:
+        - ApiKeyAuth: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Donation'
+      responses:
+        '200':
+          $ref: '#/components/responses/OkReturnId'
+  /donations/{donationId}:
+    parameters:
+      - name: donationId
+        in: path
+        required: true
+        description: Donation id (uuid).
+        schema:
+          type: integer
+          format: int64
+          minimum: 1
+    get:
+      description: Get donation by id.
+      responses:
+        '200':
+          $ref: '#/components/responses/DonationsOk'
+    put:
+      description: Put donation with client specified id (uuid).
+      security:
+        - ApiKeyAuth: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Donation'
+      responses:
+        '200':
+          $ref: '#/components/responses/OkEmpty'
+    patch:
+      description: Patch donation data.
+      security:
+        - ApiKeyAuth: []
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Donation'
+      responses:
+        '200':
+          $ref: '#/components/responses/OkEmpty'
+    delete:
+      description: Delete donation.
+      security:
+        - ApiKeyAuth: []
+      responses:
+        '200':
+          $ref: '#/components/responses/OkEmpty'
+  /gamedata:
+    get:
+      description: >
+        Get list of all game data. List of games that are scheduled for the event.
+      
+      responses:
+        '200':
+          $ref: '#/components/responses/GameDataAllOk'
+    post:
+      description: Post game data, id generated by backend.
```

I don't think it's necessary to mention the id generation. I'd prefer mentioning what the endpoint is used
```suggestion
      description: Add a new gamedata object
```